### PR TITLE
Release 4.6.0

### PR DESCRIPTION
**Pull request by asottile**
:tada: :fireworks: :tada: :fireworks: :tada:

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -18,6 +18,76 @@ with advance notice in the **Deprecations** section of releases.
 
 .. towncrier release notes start
 
+pytest 4.6.0 (2019-05-31)
+=========================
+
```
Comment (nicoddemus) [inline CHANGELOG.rst]: We should make it very clear that this is the last major series to support Python 2. I suggest:

```rst
Important
---------

The ``4.6.X`` series will be the last series to support **Python 2 and Python 3.4**. 

For more details, see our `Python 2.7 and 3.4 support plan <https://docs.pytest.org/en/latest/py27-py34-deprecation.html>`__.
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -21,6 +21,14 @@ with advance notice in the **Deprecations** section of releases.
 pytest 4.6.0 (2019-05-31)
 =========================
 
+Important
```
Comment (nicoddemus) [inline CHANGELOG.rst]: In retrospect, I could have done this myself. 😅

Comment (asottile) [reply CHANGELOG.rst]: hehehe -- I did it anyway all good!

**Comment by asottile** [comment]
@nicoddemus an idea why `codecov/patch` hasn't reported? can I ignore that and continue?

**Comment by nicoddemus** [comment]
> @nicoddemus an idea why codecov/patch hasn't reported? can I ignore that and continue?

No idea why, but let's wait for @blueyed's review regardless. 👍

**Comment by blueyed** [comment]
Codecov likely got confused (reported -4% drop at https://codecov.io/gh/pytest-dev/pytest/pull/5350/changes).
(I can imagine this happens when the merge commit advanced already or something like that (similar to how Travis does not build the actual commit to be tested when you push a new commit before it gets to this job))

File: CHANGELOG.rst [hunk 3]
```diff
@@ -18,6 +18,84 @@ with advance notice in the **Deprecations** section of releases.
...
+
+The ``4.6.X`` series will be the last series to support **Python 2 and Python 3.4**.
+
+For more details, see our `Python 2.7 and 3.4 support plan <https://docs.pytest.org/en/latest/py27-py34-deprecation.html>`__.
+
+
+Features
+--------
+
+- `#4559 <https://github.com/pytest-dev/pytest/issues/4559>`_: Added the ``junit_log_passing_tests`` ini value which can be used to enable and disable logging passing test output in the Junit XML file.
```
Comment (blueyed) [inline CHANGELOG.rst]:
```suggestion
- `#4559 <https://github.com/pytest-dev/pytest/issues/4559>`_: Added the ``junit_log_passing_tests`` ini value which can be used to enable or disable logging of passing test output in the Junit XML file.
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -18,6 +18,84 @@ with advance notice in the **Deprecations** section of releases.
...
+
+
+- `#4956 <https://github.com/pytest-dev/pytest/issues/4956>`_: pytester's ``testdir.spawn`` uses ``tmpdir`` as HOME/USERPROFILE directory.
+
+
+- `#5062 <https://github.com/pytest-dev/pytest/issues/5062>`_: Unroll calls to ``all`` to full for-loops for better failure messages, especially when using Generator Expressions.
```
Comment (blueyed) [inline CHANGELOG.rst]:
```suggestion
- `#5062 <https://github.com/pytest-dev/pytest/issues/5062>`_: Unroll calls to ``all`` to full for-loops with assertion rewriting for better failure messages, especially when using Generator Expressions.
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -18,6 +18,84 @@ with advance notice in the **Deprecations** section of releases.
...
+
+
+- `#5269 <https://github.com/pytest-dev/pytest/issues/5269>`_: ``pytest.importorskip`` includes the ``ImportError`` now in the default ``reason``.
+
+
+- `#5311 <https://github.com/pytest-dev/pytest/issues/5311>`_: Captured logs that are output for each failing test are formatted using the
+  ColoredLevelFormatter.
+
+
+- `#5312 <https://github.com/pytest-dev/pytest/issues/5312>`_: Improved formatting of multiline log messages in python3.
```
Comment (blueyed) [inline CHANGELOG.rst]:
```suggestion
- `#5312 <https://github.com/pytest-dev/pytest/issues/5312>`_: Improved formatting of multiline log messages in Python 3.
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -18,6 +18,84 @@ with advance notice in the **Deprecations** section of releases.
...
+- `#5256 <https://github.com/pytest-dev/pytest/issues/5256>`_: Handle internal error due to a lone surrogate unicode character not being representable in Jython.
+
+
+- `#5257 <https://github.com/pytest-dev/pytest/issues/5257>`_: Ensure that ``sys.stdout.mode`` does not include ``'b'`` as it is a text stream.
+
+
+- `#5278 <https://github.com/pytest-dev/pytest/issues/5278>`_: Pytest's internal python plugin can be disabled using ``-p no:python`` again.
+
+
+- `#5286 <https://github.com/pytest-dev/pytest/issues/5286>`_: Fix issue with ``disable_test_id_escaping_and_forfeit_all_rights_to_community_support`` option doesn't work when using a list of test IDs in parametrized tests.
```
Comment (blueyed) [inline CHANGELOG.rst]:
```suggestion
- `#5286 <https://github.com/pytest-dev/pytest/issues/5286>`_: Fix issue with ``disable_test_id_escaping_and_forfeit_all_rights_to_community_support`` option not working when using a list of test IDs in parametrized tests.
```

Comment (blueyed) [reply CHANGELOG.rst]: Given its name etc it should probably not show up changelogs?!  Anyway..

File: CHANGELOG.rst [hunk 7]
```diff
@@ -18,6 +18,84 @@ with advance notice in the **Deprecations** section of releases.
...
+
+
+- `#5330 <https://github.com/pytest-dev/pytest/issues/5330>`_: Show the test module being collected when emitting ``PytestCollectionWarning`` messages for
+  test classes with ``__init__`` and ``__new__`` methods to make it easier to pin down the problem.
+
+
+- `#5333 <https://github.com/pytest-dev/pytest/issues/5333>`_: Fix regression with ``--lf`` not re-running all tests with known failures from non-selected tests.
```
Comment (blueyed) [inline CHANGELOG.rst]:
```suggestion
- `#5333 <https://github.com/pytest-dev/pytest/issues/5333>`_: Fix regression in 4.5.0 with ``--lf`` not re-running all tests with known failures from non-selected tests.
```

**Comment by blueyed** [comment]
I'd say to wait for CI to pass before rebasing this for fixed changelogs (with regard to codecov).

**Comment by nicoddemus** [comment]
Ready for the release! 🎉

**Pull request merged**